features/profile_screen/ProfileScreen.tsx

```diff
@@ -0,0 +1,150 @@
+import { Image, Pressable, Text, View } from "react-native";
+import React, { useMemo, useState } from "react";
+import { styles } from "./styles";
+import { PersonalDataWidget } from "./PersonalDataWidget/PersonalDataWidget";
+import { useSelector } from "react-redux";
+import { profileSelector } from "../../store/features/profile/selectors";
+import { WalletsList } from "./WalletsList/WalletsList";
+import { IWallet } from "./Wallet/types";
+import { SortingParam } from "../../enum/wallets_sorting/SortingParam";
+import { Sorted } from "../../enum/wallets_sorting/Sorted";
+
+const createSortingFunction = (param: SortingParam, sorted: Sorted) => {
+  if (param === SortingParam.CODE) {
+    return (sorted === Sorted.FIRST) ? sortByCodeMinToMax : sortByCodeMaxToMin
+  }
+  if (param === SortingParam.VOLUME) {
+    return (sorted === Sorted.FIRST) ? sortByVolumeMaxToMin : sortByVolumeMinToMax
+  }
+  return sortByCodeMinToMax;
+}
+
+const sortByVolumeMinToMax = (a: IWallet, b: IWallet) => {
+  return a.volume - b.volume;
+}
+
+const sortByVolumeMaxToMin = (a: IWallet, b: IWallet) => {
+  return b.volume - a.volume;
+}
+
+const sortByCodeMinToMax = (a: IWallet, b: IWallet) => {
+  if (a.code < b.code) {
+    return -1;
+  }
+  if (a.code > b.code) {
+    return 1;
+  }
+  return 0;
+}
+
+const sortByCodeMaxToMin = (a: IWallet, b: IWallet) => {
+  if (a.code < b.code) {
+    return 1;
+  }
+  if (a.code > b.code) {
+    return -1;
+  }
+  return 0;
+}
+
+
+const ProfileScreen = () => {
+
+  const [sortingParam, setSortingParam] = useState<SortingParam>(SortingParam.CODE);
+  const [sorted, setSorted] = useState<Sorted>(Sorted.NOT_SORTED);
+
+  const profile = useSelector(profileSelector);
+
+  const sortedWallets = useMemo(() => {
```

Comment:
Еще наверно не говорил, поэтому отмечу сейчас:
если код компонента начинает разрастаться, то лучше выносить хуки в отдельные функции. То есть тут надо все useMemo  вынести в хук useSortedWallets. Тут это не важно, конечно, потому что кода в компоненте не очень много, но вынести всегда точно не хуже, чем не вынести)

Reply:
Также в хуки можно выносить (и нужно) группировки других хуков (когда они делают что-то общее по смыслу, и адекватнымшагом видится их объединение). То есть получится какая-то такая функция, например:
`
```suggestion
  const {
    sortedWallets,
    callback
  } = useSortedWallets(...)
  
  ....
  
  const useSortedWalltes = (...) => {
   const sortedWallters = useMemo(...)
   const collback = useCallback(...) // ЭТО тут чисто как пример, если бы, например, было нужно еще и колбек выкинуть наружу, который, например, меняет порядок сортировки
   useEffect()
   
   return {sortedWallters , callback};
 }
```